app/services/fees/charge_service.rb

```diff
@@ -35,7 +35,7 @@ def call
           precise_amount_cents: result.fees.sum(&:precise_amount_cents)
         )
       end
-      return result unless result.success?
+      return result if !result.success? || context == :invoice_preview
```

Comment:
Do we want to render zero-amount fees in the preview? If we escape here, the service will return the full list of fees, including the zero amount ones

Reply:
Hmm good question? I guess both options are fine. The idea was to escape here so that we still create true up fee if needed, but indeed, zero fees will be included.
Do you suggest to exclude all zero amount fees from the array before returning?

Reply:
Maybe it's better to be consistent with the latest changes. What do you think?

Reply:
okay, moved `return` after `result.fees.reject! { |f| !should_persit_fee?(f, result.fees) }` line 👍